snuba/settings.py

```diff
@@ -69,9 +69,23 @@
 BULK_CLICKHOUSE_BUFFER = 10000
 
 # Processor/Writer Options
-DEFAULT_BROKERS = ["localhost:9092"]
+DEFAULT_BROKERS: Sequence[str] = os.environ.get(
+    "DEFAULT_BROKERS", "localhost:9092"
+).split(",")
 DEFAULT_STORAGE_BROKERS: Mapping[str, Sequence[str]] = {}
 
+_BROKER_CONFIG: Mapping[str, Any] = {
+    "bootstrap.servers": ",".join(DEFAULT_BROKERS),
+    "sasl.mechanism": os.environ.get("SASL_MECHANISM"),
+    "sasl.username": os.environ.get("SASL_PLAIN_USERNAME"),
+    "sasl.password": os.environ.get("SASL_PLAIN_PASSWORD"),
+    "security.protocol": os.environ.get("SECURITY_PROTOCOL"),
+}
+BROKER_CONFIG: Mapping[str, Any] = {
+    k: v for k, v in _BROKER_CONFIG.items() if v is not None
+}
+STORAGE_BROKER_CONFIG: Mapping[str, Mapping[str, Any]] = {}
```

Comment:
But as far as I understand the settings.py load order, at the time this execute, the use settings.py is not yet loaded. 

So my STORAGE_BROKER_CONFIG will always be empty, as DEFAULT_STORAGE_BROKERS is always empty in the default settings.py that is loaded before the local settings.py.

I can probably postpone the execution of the code that computes the value of STORAGE_BROKER_CONFIG  to the time it is first read, but it really doesn't feel like a proper solution.

What do you suggests with regards to load order?

Reply:
I think you are right, and this impacts DEFAULT_BROKER/BROKER_CONFIG. IF somebody overrides DEFAULT_BROKER, BROKER_CONFIG won't see the change. 
I guess there is no alternative other than keep the variables (DEFAULT_BROKER/BROKER_CONFIG and STORAGE_BROKER_CONFIG/DEFAULT_STORAGE_BROKERS) separate.

1) this PR would leave the old variables there empty
2) it would move the config in the new variables
3) the code (see the comment where I advise to move the config loading code in one place only) will read first the old variable, then, if empty the new one.
4) When enough time will have passed that we can decommission the old one we will remove that code_path.